docs/prover/diagnosis/index.md

```diff
@@ -5,15 +5,238 @@ Diagnostic Tools
+inlinings and optimizations already applied, it is very close to what the SMT
+solver sees. 
+
+Some commands in the TAC source code have pointers to the program or
+specification code that the TAC program was created from. These commands are
+highlighted a dotted underline; the source code location they have been
+translated from is revealed on hovering with the mouse over such a command, as
+displayed in the image below.
+
+
+```{figure} source-pointer-on-hover-annotated.png
+:name: source-pointer-on-hover
+:alt: Source pointer shown on hover
+:align: center
+
+source pointer is shown on hover over TAC commands that have a dotted underline
+```
+
+The call-based navigation on the mid-right side of the report lists all the
+calls in the TAC program. Each call is identified by a number. After the number
+there is a number of arrows, indicating the nesting depth of the call, i.e.
+calls from spec get one arrow, a call from a method body with one arrow gets two
+arrows, etc.; see the picture below for an example. After the arrows the name of
+the called method is given. Each of these calls is clickable and will lead to
+the corresponding CFG.
+
+```{figure} call-depth-arrows.png
+:name: call-depth-arrows
+:alt: Arrows indicating depth of call
+:align: center
+:height: 300
+
+arrows indicating the "depth" of a call in the call links
+```
+
+% TODO: ok to show these Aave/delvtech|element method names in this picture?
+
+At the bottom of the TAC report are the successor and predecessor relations of
+the CFG on a per-block/per-node basis. These are usually not useful for the tool
+user, so we will ignore them in the remainder of this document.
+
+### SAT TAC reports
+
+Once a rule has been checked with a {term}`SAT` outcome, a TAC report like the one
+in the figure below is generated. In addition to the components of the plain TAC
+report, this report variant illustrates the {term}`model` (aka. counterexample)
+that is provided by the SMT solver to witness violation of an `assert` command
+(or the fulfillment of a `satisfy` command). The model is illustrated by a path
+in the control flow graph and a valuation of the TAC program variables.
+
+
+```{figure} sat-tac-report-plain.png
+:name: sat-tac-report
+:align: center
+
+example TAC report for a run with a SAT result
+```
+
+The path corresponds to the {term}`call trace` from the
+[report](verification-report). It is the path through the program that is taken
+on the input given by the counterexample. The arrows and nodes in the control
+flow graph that lie on this path are displayed in red.
+
+The valuation is given in the box in the top right. It lists every variable in
+the TAC program along with the value that it has in the current counterexample.
+
+
+```{note}
+The names of the TAC variables are often not helpful for understanding what they 
+mean. The valuation can nonetheless be useful for variables whose meaning is 
+clear from context (e.g. operations they occur in, or the source pointers).
```

Comment:
It might be worth mentioning `--prover_args "-canonicalizeTAC false"` here. It might prevent the cache from being used and also make the performance of the prover a bit flaky, however, it preserves the CVL names and hence make it more readable.

Reply:
added a sentence